internal/controller/authenticator/webhookcachefiller/webhookcachefiller.go

```diff
@@ -428,11 +430,11 @@ func (c *webhookCacheFillerController) validateConnection(
 		return conditions, nil
 	}
 
-	conn, err := tls.Dial("tcp", endpointHostPort.Endpoint(), ptls.Default(certPool))
+	err := c.dialer.IsReachableAndTLSValidationSucceeds(endpointHostPort.Endpoint(), certPool, logger)
```

Comment:
we could perhaps follow the same pattern that is used in `upstream_watchers.go` where we create a context w/ a timeout and pass that context to the validation function. And the validation function could then use that context to decide the timeout value.